Sources/Sham/Models/StubRequest.swift

```diff
@@ -43,23 +52,33 @@ public struct StubRequest: Hashable, CustomStringConvertible, Codable {
     // MARK: Initializers
     
     /// Initializes a StubRequest.
-    /// - Parameter method: The HTTP method to stub.
-    /// - Parameter url: The URL to stub.
-    public init(method: HTTPMethod, url: URLConvertible) {
+    /// - Parameters:
+    ///   - method: The HTTP method to stub.
+    ///   - url: The URL to stub.
+    ///   - validationRule: The validation rule to use when comparing requests.
+    public init(method: HTTPMethod,
+                url: URLConvertible,
+                validationRule: StubRequestValidationRule = .allowMissingQueryParameters) {
```

Comment:
Before this change, wasn't validation requiring an exact match? If so, we may want to default any options to `.explicit` instead to avoid breaking changes.